EIPS/eip-888.md

```diff
@@ -0,0 +1,178 @@
+    function transfer(address _to, uint256 _amount) external;
+
+    /// @notice Transfer EXP from one address to another.
+    /// @dev MUST throw unless msg.sender is operator.
+    ///  MUST throw unless _to address is participating.
+    ///  MAY throw if _from address is NOT participating.
+    function transferFrom(address _from, address _to, uint256 _amount) external;
+
+    /// @notice Activate or deactivate participation.
+    /// @dev MUST throw unless msg.sender is _participant.
+    ///  MUST throw if _participant is _operator or zero address.
+    /// @param _participant Address opting in or out of participation.
+    /// @param _participation Participation status of _participant.
+    function approve(address _participant, bool _participation) external;
+
+    /// @notice Reassign operator authority.
+    /// @dev MUST throw unless msg.sender is _operator.
+    ///  MUST if _operator address is already current _operator.
+    /// @param _operator New operator of the smart contract.
+    function setOperator(address _operator) external;
+}
+```
+
+The *metadata extension* is OPTIONAL for ERC-888 smart contracts. This allows an EXP smart contract to be interrogated for its name and description.
+```
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./IERC888.sol";
+
+/// @title ERC-888 EXP Standard, optional metadata extension
+/// @dev See https://eips.ethereum.org/EIPS/EIP-888
+///  Note: the ERC-165 identifier for this interface is ###ERC888Metadata###.
+interface IERC888Metadata is IERC888 {
+    /// @notice A descriptive name for the EXP in this contract.
+    function name() external view returns (string memory);
+
+    /// @notice A one-line description of the EXP in this contract.
+    function description() external view returns (string memory);
+}
+```
+
+## Rationale
+### Approval
+EXP drops SHALL require pre-approval from the delivery address. This ensures the receiver is a consenting participant in the smart contract.
+
+### Mints & Transfers
+EXP mints and transfers SHALL be at the soul discretion of the contract operator. This party may be a sports team coach or a multisig DAO wallet. We decide not to specify how governance occurs, but only *that* governance occurs. This allows for a wider range of potential use cases than optimizing for particular decision-making forms.
+
+ERC-888 standardizes a control mechanism to allocate community recognition without encouraging financialization of that recognition or easily allowing non-contributors to acquire EXP representing contribution. While it does not ensure meritocracy, it opens the door.
+
+### Token Destruction
+EXP SHOULD allow burning tokens by contract operators. If Bob has contributed greatly to the community, but then is caught stealing from Alice, the community may decide this should lower Bob's standing and influence in the community. Again, while this does not ensure an ethical standard within the community, it opens the door.
```

Comment:
I'd recommend codifying the `burn` function in the specification while taking out `transferFrom`

Reply:
Currently, `transferFrom` allows moving EXP tokens from one participant to another or to the zero address. `transfer` is specified as coming from the zero address. In my mind, I was trying to align closely with 20 and 721, but maybe the use case is too different and/or it's just too confusing in the EXP token context.

A more intuitive solution could be:
- `mint` to a not-zero address
- `burn` to the zero address
- `transfer` from a not-zero address to another not-zero address
- Take out `transferFrom` altogether

Each function would have a corresponding event as well.

Is that sensible?

Reply:
What about the invariant that the EXP cannot be transferred between addresses?

Why not only:
- `mint` to a non-zero address
- `burn` to the zero address, amount specifiable

Reply:
Fair question. The key is non-tradability rather than non-transferability. `transfer` is important for any scarcity mechanic as well as a few other potential use cases:
- In a video game, if two characters battle, contract operator transfers some EXP from the loser to the winner
- If it's discovered EXP was allocated to the wrong participant, contract operator can rectify.
- If a participant changes addresses, contract operator can move over the EXP.
- If distribution of EXP becomes unbalanced over time, contract operator can redistribute.

Of course, we can reach the same state using only `mint` and `burn`, but that requires more transactions and casuality would harder to infer.

Does that make sense? However we decide, I'll make sure to add reasoning for this decision in the EIP.

Reply:
I see. The distinction between non-tradability is very important. Then I agree a `transferFrom` and `_transfer` function are needed, and we could implement like this:

All must be called by operator
- `mint`: calls internal `_transfer`, must be from 0x0 to participating EOA or contract address 
- `burn`: calls internal `_transfer`, must be to 0x0 from a participating EOA or contract address where amount <= balanceOf(account)
- `transferFrom`: calls internal `_transfer`, both addresses must be participating. No approval required by the `_from` account.

I think that would cover most use cases.

Reply:
This makes sense to me. Thanks. 

In the latest commit, I moved in this direction, but will finalize and test before next review.

Reply:
Further discussion on this topic on the [Ethereum Magicians thread](https://ethereum-magicians.org/t/eip-4974-fungible-non-tradable-tokens-or-exp/8805/10).